src/components/engagement/handlers/update-project-status.handler.ts

```diff
@@ -53,4 +54,52 @@ export class UpdateProjectStatusHandler
       })
     );
   }
+
+  async handle({ previous, updated, updates, session }: ProjectUpdatedEvent) {
+    if (updated.status === ProjectStatus.Terminated) {
+      // filter out any engagements that have already reached their terminus
+      const engagements = (await this.getEngagements(updates.id)).filter(
```

Comment:
maybe `nonTerminalEngagements` would be clearer

Reply:
I just wanted to use terminus in a sentence